x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/expression/function/scalar/math/ToNumberFunctionProcessorTests.java

```diff
@@ -19,12 +21,17 @@ private static Object process(Object value, Object base) {
         return new ToNumber(EMPTY, l(value), l(base)).makePipe().asProcessor().process(null);
     }
 
+    private static String error(Object value, Object base, Class<? extends QlException> exceptionClass) {
+        Exception e = expectThrows(exceptionClass, () -> new ToNumber(EMPTY, l(value), l(base)).makePipe().asProcessor().process(null));
+        return e.getMessage();
+    }
+
     private static String error(Object value, Object base) {
-        QlIllegalArgumentException saie = expectThrows(
-            QlIllegalArgumentException.class,
-            () -> new ToNumber(EMPTY, l(value), l(base)).makePipe().asProcessor().process(null)
-        );
-        return saie.getMessage();
+        return error(value, base, QlIllegalArgumentException.class);
+    }
+
+    private static String clientError(Object value, Object base) {
+        return error(value, base, EqlClientException.class);
     }
```

Comment:
nit: Similarly here, we could also require a verification exception since all usages of `clientError` fall under this category IMO.